authz/rbac_translator_test.go

```diff
@@ -0,0 +1,231 @@
+											UrlPath: &v3matcherpb.PathMatcher{Rule: &v3matcherpb.PathMatcher_Path{Path: &v3matcherpb.StringMatcher{
+												MatchPattern: &v3matcherpb.StringMatcher_Suffix{Suffix: "baz"}}}}}},
+									}}}},
+								{Rule: &v3rbacpb.Permission_AndRules{AndRules: &v3rbacpb.Permission_Set{
+									Rules: []*v3rbacpb.Permission{
+										{Rule: &v3rbacpb.Permission_OrRules{OrRules: &v3rbacpb.Permission_Set{
+											Rules: []*v3rbacpb.Permission{
+												{Rule: &v3rbacpb.Permission_Header{
+													Header: &v3routepb.HeaderMatcher{
+														Name: "key-1", HeaderMatchSpecifier: &v3routepb.HeaderMatcher_ExactMatch{ExactMatch: "foo"}}}},
+												{Rule: &v3rbacpb.Permission_Header{
+													Header: &v3routepb.HeaderMatcher{
+														Name: "key-1", HeaderMatchSpecifier: &v3routepb.HeaderMatcher_SuffixMatch{SuffixMatch: "bar"}}}},
+											}}}},
+										{Rule: &v3rbacpb.Permission_OrRules{OrRules: &v3rbacpb.Permission_Set{
+											Rules: []*v3rbacpb.Permission{
+												{Rule: &v3rbacpb.Permission_Header{
+													Header: &v3routepb.HeaderMatcher{
+														Name: "key-2", HeaderMatchSpecifier: &v3routepb.HeaderMatcher_PrefixMatch{PrefixMatch: "baz"}}}},
+											}}}}}}}}}}}}},
+				},
+			}},
+		},
+		"missing name field": {
+			authzPolicy:     `{}`,
+			wantErr:         "\"name\" is not present",
+			wantDenyPolicy:  nil,
+			wantAllowPolicy: nil,
+		},
+		"invalid field type": {
+			authzPolicy:     `{"name": 123}`,
+			wantErr:         "failed to unmarshal policy",
+			wantDenyPolicy:  nil,
+			wantAllowPolicy: nil,
+		},
+		"missing allow rules field": {
+			authzPolicy:     `{"name": "authz-foo"}`,
+			wantErr:         "\"allow_rules\" is not present",
+			wantDenyPolicy:  nil,
+			wantAllowPolicy: nil,
+		},
+		"missing rule name field": {
+			authzPolicy: `{
+				"name": "authz-foo",
+				"allow_rules": [{}]
+			}`,
+			wantErr:         "\"allow_rules\" 0: \"name\" is not present",
+			wantDenyPolicy:  nil,
+			wantAllowPolicy: nil,
+		},
+		"missing header key": {
+			authzPolicy: `{
+				"name": "authz",
+				"allow_rules": [{"name": "allow_policy_1", "request": {"headers":[{}]}}]
+			}`,
+			wantErr:         "\"allow_rules\" 0: \"headers\" 0: \"key\" is not present",
+			wantDenyPolicy:  nil,
+			wantAllowPolicy: nil,
+		},
+		"missing header values": {
+			authzPolicy: `{
+				"name": "authz",
+				"allow_rules": [{"name": "allow_policy_1", "request": {"headers":[{"key":"key-a"}]}}]
+			}`,
+			wantErr:         "\"allow_rules\" 0: \"headers\" 0: \"values\" is not present",
+			wantDenyPolicy:  nil,
+			wantAllowPolicy: nil,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			gotDenyPolicy, gotAllowPolicy, gotErr := translatePolicy(test.authzPolicy)
+			if gotErr != nil && !strings.HasPrefix(gotErr.Error(), test.wantErr) {
+				t.Fatalf("translatePolicy%v\nunexpected error\nwant:%v\ngot:%v", test.authzPolicy, test.wantErr, gotErr)
```

Comment:
nit: for all the `t.Fatalf` calls, it's probably not needed to include test.authzPolicy(or at least it can be replaced with `name`), since  `t.Run(name, func(t *testing.T)` will already log the name of the errored test cases, IIRC.